webui/react/src/components/ParallelCoordinates.tsx

```diff
@@ -199,15 +204,15 @@ const ParallelCoordinates: React.FC<Props> = ({
     });
 
     return () => {
-      if (ref) Plotly.purge(ref);
+      Plotly.purge(ref);
     };
     /* eslint-disable-next-line react-hooks/exhaustive-deps */
```

Comment:
Since that's an anti-pattern we should (?) avoid, what do you think about adding a small comment to document why that's needed? (there are other occurrences in <ScatterPlot>)

Reply:
I tried to understand the reason for skipping exaustive deps, is that mainly related to:
- uplot "watching" the chartData (no need to trigger a refresh)
- not wanting to re-generate the chart but needing to pass the data on the initialization?

I think that can be solved in 2 ways:
1. generating chartData in a useEffect and storing into a useRef (changing the value will not trigger the useeffect).
2. providing no data to uplot on init and (i believe that's possible) call a method on the uplot instance to update the chart data?

Reply:
> I tried to understand the reason for skipping exaustive deps, is that mainly related to:
> 
> * uplot "watching" the chartData (no need to trigger a refresh)
> * not wanting to re-generate the chart but needing to pass the data on the initialization?
> 
> I think that can be solved in 2 ways:
> 
> 1. generating chartData in a useEffect and storing into a useRef (changing the value will not trigger the useeffect).
> 2. providing no data to uplot on init and (i believe that's possible) call a method on the uplot instance to update the chart data?

Very good points, and would love to chat about this in the WebUI AG. Curious if you have been able to avoid skipping the exhaustive dependencies linting completely.